### PR TITLE
Subcommand derive-pk

### PR DESCRIPTION
**Pull request by kpp**

### CODE REVIEW

File: src/node_config.rs [hunk 1]
```diff
@@ -160,6 +160,25 @@ fn app() -> App<'static, 'static> {
                 .help("Load settings from saved config file. \
                     Config file format is YAML")
                 .takes_value(true)))
+        .subcommand(SubCommand::with_name("get_pk")
```
Comment (kurnevsky) [inline src/node_config.rs]: I think hyphen'd be better: https://stackoverflow.com/questions/1253679/should-command-line-options-in-posix-style-operating-systems-be-underscore-style

Comment (kurnevsky) [reply src/node_config.rs]: Also how about `derive` instead of `get`?

File: src/node_config.rs [hunk 2]
```diff
@@ -283,6 +297,36 @@ fn parse_config(config_path: &str) -> NodeConfig {
...
 }
 
+fn run_derive_pk(matches: &ArgMatches) -> ! {
+    let sk_passed_as_arg = matches.occurrences_of("secret-key") > 0;
+    if sk_passed_as_arg {
+        warn!("You should not pass the secret key via arguments due to \
+               security reasons. Use the environment variable instead");
+    }
+
+    let sk_from_arg = matches.value_of("secret-key").map(|s| {
```
Comment (kurnevsky) [inline src/node_config.rs]: How about:
```diff
diff --git a/src/node_config.rs b/src/node_config.rs
index b76d338..4840c5c 100644
--- a/src/node_config.rs
+++ b/src/node_config.rs
@@ -304,11 +304,12 @@ fn run_derive_pk(matches: &ArgMatches) -> ! {
                security reasons. Use the environment variable instead");
     }

-    let sk_from_arg = matches.value_of("secret-key").map(|s| {
+    let pk_from_arg = matches.value_of("secret-key").map(|s| {
         let sk_bytes: [u8; 32] = FromHex::from_hex(s).expect("Invalid DHT secret key");
-        SecretKey::from_slice(&sk_bytes).expect("Invalid DHT secret key")
+        let sk = SecretKey::from_slice(&sk_bytes).expect("Invalid DHT secret key");
+        sk.public_key()
     });
-    let sk_from_file = matches.value_of("keys-file").map(|keys_file| {
+    let pk_from_file = matches.value_of("keys-file").map(|keys_file| {
         let mut file = std::fs::File::open(keys_file).expect("Failed to read the keys file");

         let mut buf = [0; PUBLICKEYBYTES + SECRETKEYBYTES];
@@ -317,12 +318,12 @@ fn run_derive_pk(matches: &ArgMatches) -> ! {
         let pk = PublicKey::from_slice(&buf[..PUBLICKEYBYTES]).expect("Failed to read public key from the keys file");
         let sk = SecretKey::from_slice(&buf[PUBLICKEYBYTES..]).expect("Failed to read secret key from the keys file");
         assert!(pk == sk.public_key(), "The loaded public key does not correspond to the loaded secret key");
-        sk
+        pk
     });

-    let sk = sk_from_arg.or(sk_from_file).unwrap();
+    let pk = pk_from_arg.or(pk_from_file).unwrap();

-    println!("{}", hex::encode(sk.public_key().as_ref()).to_uppercase());
+    println!("{}", hex::encode(pk.as_ref()).to_uppercase());

     std::process::exit(0)
 }
```

File: src/node_config.rs [hunk 3]
```diff
@@ -283,6 +297,37 @@ fn parse_config(config_path: &str) -> NodeConfig {
...
+               security reasons. Use the environment variable instead");
+    }
+
+    let pk_from_arg = matches.value_of("secret-key").map(|s| {
+        let sk_bytes: [u8; 32] = FromHex::from_hex(s).expect("Invalid DHT secret key");
+        let sk = SecretKey::from_slice(&sk_bytes).expect("Invalid DHT secret key");
+        sk.public_key()
+    });
+    let pk_from_file = matches.value_of("keys-file").map(|keys_file| {
+        let mut file = std::fs::File::open(keys_file).expect("Failed to read the keys file");
```
Comment (kpp) [inline src/node_config.rs]: Should I move it to a standalone function or is ^C^V fine?

File: src/node_config.rs [hunk 4]
```diff
@@ -283,6 +297,37 @@ fn parse_config(config_path: &str) -> NodeConfig {
     config
 }
 
+fn run_derive_pk(matches: &ArgMatches) -> ! {
+    let sk_passed_as_arg = matches.occurrences_of("secret-key") > 0;
+    if sk_passed_as_arg {
+        warn!("You should not pass the secret key via arguments due to \
```
Comment (kurnevsky) [inline src/node_config.rs]: Log backend isn't initialized here yet so this warn is invisible.

File: src/node_config.rs (hunk 3 above)
Comment (kurnevsky) [reply src/node_config.rs]: It won't harm if you can :)
Maybe also move it to main.rs to solve the problem with log?

Comment (kpp) [reply src/node_config.rs]: Let's do this in a separate PR

**Pull request merged**